ksql-examples/src/main/java/io/confluent/ksql/datagen/DataGenProducer.java

```diff
@@ -60,9 +67,11 @@ public void populateTopic(
       maxInterval = INTER_MESSAGE_MAX_INTERVAL;
     }
     Schema avroSchema = generator.schema();
-    org.apache.kafka.connect.data.Schema kafkaSchema = new AvroData(1).toConnectSchema(avroSchema);
+    final AvroData avroData = new AvroData(1);
+    org.apache.kafka.connect.data.Schema ksqlSchema = avroData.toConnectSchema(avroSchema);
+    ksqlSchema = getOptionalSchema(ksqlSchema);
```

Comment:
Why do we need to have an optional schema here? KSQL should work whether or not the schema in the registry has optional fields. And the _serializers_ shouldn't require optional schemas. They just fill in the struct and hand off to the the connect API. Whoever is creating the serializer only needs to ensure fields are optional if they can take null values. But in this case the random generator should ensure that nulls are only generated if the schema expects nulls.

Reply:
This one is being used to generate optional values for the rows. These rows that can include struct columns are generated by the data gen, not the serializer.

Reply:
Right, and data gen shouldn't generate null for rows that aren't optional in the original schema. So you shouldn't need to pass the serializer a schema that has everything set to null. Does it not work without this?

Reply:
It doesn't work without it. In addition we don't support `NOT NULL` property for columns so regardless we can always have NULL columns. We can look into removing this requirement in another PR.